parquet-column/src/main/java/org/apache/parquet/column/values/dictionary/DictionaryValuesWriter.java

```diff
@@ -173,7 +173,7 @@ public BytesInput getBytes() {
       BytesInput bytes = concat(BytesInput.from(bytesHeader), rleEncodedBytes);
       // remember size of dictionary when we last wrote a page
       lastUsedDictionarySize = getDictionarySize();
-      lastUsedDictionaryByteSize = dictionaryByteSize;
+      lastUsedDictionaryByteSize = (int) dictionaryByteSize;
```

Comment:
This method should not be called in case of `shouldFallBack()` returns true be we would better be on the safe side and check the potential overflow.

Reply:
Okay. Changed to `Math.toIntExact` so it will throw exception when overflow.

Reply:
If that user continues writing that large string, will here throw an exception?  And it will block the writer, right?

Reply:
I don't think so. With the fix, it should've already fallback to PLAIN encoding during `writeBytes` before getting here.